File: src/main/java/application/history/history/domain/entities/History.java
package application.history.history.domain.entities;

import lombok.Getter;
import lombok.NoArgsConstructor;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.Table;
import java.text.SimpleDateFormat;
import java.time.LocalDate;

@NoArgsConstructor
@Entity
@Getter
@Table(name = "history")
public class History {
    @Id
    @Column(columnDefinition = "VARCHAR(36)")
    private String id;

    @Column
    private String content;

    @Column(columnDefinition = "DATE")
    private LocalDate date;

    public History(String id, String content, LocalDate date) {
        this.id = id;
        this.content = content;
        this.date = date;
    }

    public String getDateByFormat(String format) {
        SimpleDateFormat simpleDateFormat = new SimpleDateFormat(format);

        return simpleDateFormat.format(this.date);
    }

    public void updateContent(String content) {
        this.content = content;
    }
}

File: src/history/main/kotlin/com/example/history/domain/entities/History.kt
package com.example.history.domain.entities

import java.text.SimpleDateFormat
import java.time.LocalDate
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.Id
import javax.persistence.Table

@Entity
@Table(name = "history")
class History(
    @field:Column(columnDefinition = "VARCHAR(36)") @field:Id private val id: String,
    @field:Column private var content: String,
    date: LocalDate
) {
    @Column(columnDefinition = "DATE")
    private val date: LocalDate
    fun getDateByFormat(format: String?): String {
        val simpleDateFormat = SimpleDateFormat(format)
        return simpleDateFormat.format(date)
    }

    fun updateContent(content: String) {
        this.content = content
    }

    init {
        this.date = date
    }
}
